sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1561,6 +1562,62 @@ def plot_loading_results(self, subs=None):
 
         return PlotGrid(4, 1, ax1, ax2, ax3, ax4)
 
+    def solve_for_ild_reactions(self, val, *reactions):
+        """
+
+        Determines the Influence Line Diagram equations for reaction forces under the effect
+        of a moving load and returns a dictionary."
+
+
+        Parameters
+        ==========
+        val : Integer
+            Magnitude of moving load
+        reactions :
+            The reaction forces applied on the beam.
+
+        """
+        x = self.variable
+        l = self.length
+        C3 = Symbol('C3')
+        C4 = Symbol('C4')
+        X = symbols('X')
```

Comment:
What is the purpose of this `X` when we have `self.variable`?

Reply:
In order to solve the slope and moment equations, a limit (lim x->L) is applied. Here 'x' represents the length of the beam. The uppercase 'X' represents the distance of the moving load from origin. I have also explained this in my gsoc proposal.
![Screenshot (6)](https://user-images.githubusercontent.com/50932406/125154115-08811580-e176-11eb-828f-9d8a13c4e216.png)

Reply:
Firstly the big `X` is something the user never defined and thus would never know what it is. So you might have to provide user information about this `X` or create a similar flow of how the small `x` is taken as a parameter when a Beam object is defined.

Secondly, according to what I understand, `x` and `X` both vary over the beam length, and represent any random cross-section over the length, so we can use the same variable for them, unless there is some issue in solving the equations.

Reply:
> Secondly, according to what I understand, `x` and `X` both vary over the beam length, and represent any random cross-section over the length, so we can use the same variable for them, unless there is some issue in solving the equations.

The only reason I added 'X' was to avoid confusion as X is not any distance on the beam, it is the distance at which the moving load is acting. But if it causes more confusion then it can be definitely removed.